code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -440,3 +440,41 @@
 		M.emote(pick("twitch","laugh","frown"))
 	..()
 	. = 1
+
+/datum/reagent/drug/pumpup
+	name = "Pump-Up"
+	description = "Take on the world!"
+	reagent_state = LIQUID
+	color = "#e38e44"
+	overdose_threshold = 10
+
+/datum/reagent/drug/pumpup/on_mob_metabolize(mob/living/L)
+	..()
+	ADD_TRAIT(L, TRAIT_STUNRESISTANCE, type)
+
+/datum/reagent/drug/pumpup/on_mob_delete(mob/living/L)
```

Comment:
This currently means that the drug will be effective even if your liver is shut down or you're in stasis. You may want to consider this change:
```suggestion
/datum/reagent/drug/pumpup/on_mob_end_metabolize(mob/living/L)
```